apus_token/mint.lua

```diff
@@ -117,32 +125,48 @@ Mint.mint = function(msg)
 
         -- Retrieve the list of users eligible for minting
         local deposits = Deposits:getToAllocateUsers()
-        if not deposits or #deposits == 0 then
+        if not deposits or #deposits == 0 and not TotalMintIsZero then
             print("No users in the pool.")
             return "No users in the pool."
         end
 
-        -- Perform minting for the calculated number of times
-        for i = 1, times do
-            -- Determine the amount to be minted in this iteration
-            local releaseAmount = Mint.currentMintAmount()
-            -- Compute the reward for each user based on their deposit
-            local depositsWithReward = Allocator:compute(deposits, releaseAmount)
-            -- Update each user's balance with the calculated reward
-            Utils.map(function(r)
-                Balances[r.Recipient] = BintUtils.add(Balances[r.Recipient] or "0", r.Reward)
-            end, depositsWithReward)
-            -- Update the total minted supply by summing all user balances
-            MintedSupply = Utils.reduce(function(acc, v)
-                return BintUtils.add(acc, v)
-            end, "0", Utils.values(Balances))
+        if not deposits or #deposits == 0 and TotalMintIsZero then
+            -- reset the flag
+            TotalMintIsZero = false
+            -- total mint is zero, which results in no eligible users. Actually the process received mint reports.
+            for i = 1, times do
+                -- Determine the amount to be minted in this iteration
+                local releaseAmount = Mint.currentMintAmount()
+                -- burn the token by adding amount for balance of DEAD address
+                Balances["DEAD"] = BintUtils.add(Balances["DEAD"] or "0", releaseAmount)
```

Comment:
Might have to be unified with Token.burn
which means either Token.burn be transfer to DEAD Address.
or call Token.burn here.

Reply:
Token.burn is a handler function, should we make a message to call the function?
Changed the implementation in Token.burn, too.

Reply:
> Might have to be unified with Token.burn which means either Token.burn be transfer to DEAD Address. or call Token.burn here.

Token.burn is a handler function, should we make a message to call the function?
Changed the implementation in Token.burn, too.